dev: C# ADO.NET Docs (#17202)
ydb-platform/ydb: ydb/docs/en/core/reference/languages-and-apis/ado-net/connection_parameters.md

> # Connection parameters {#connection-parameters}

Reviewer: Mention ADO.NET in each article name. It helps people understand what they are reading without looking at the sidebar, as well as with SEO.

> To connect to a database, the application provides a connection string which specifies parameters such as the host, the user, the password, etc. Connection strings have the form `keyword1=value; keyword2=value;`. For more information, [see the official doc page on connection strings](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/connection-strings).
> 
> All available connection parameters are defined as properties in the `YdbConnectionStringBuilder`.
> 
> Below are the connection string parameters which Ydb.Sdk.Ado understands.

Reviewer: ```suggestion
To connect to a database, the application provides a connection string that specifies parameters such as the host, user, password, and so on. Connection strings have the form `keyword1=value; keyword2=value;`. For more information, [see the official doc page on connection strings](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/connection-strings).

All available connection parameters are defined as properties in the `YdbConnectionStringBuilder`.

Below are the connection string parameters that `Ydb.Sdk.Ado` understands.
```

## Basic connection

| Parameter         | Description                                                                                                 | Default value |
|-------------------|-------------------------------------------------------------------------------------------------------------|---------------|
| `Host`            | Specifies the {{ ydb-short-name }} server host                                                              | `localhost`   |

> | `Port`            | Specifies the {{ ydb-short-name }}  server port                                                             | `2136`        |

Reviewer: ```suggestion
| `Port`            | Specifies the {{ ydb-short-name }} server port                                                             | `2136`        |
```

| `Database`        | Specifies the database name                                                                                 | `/local`      |
| `User`            | Specifies the username                                                                                      | Not defined   |
| `Password`        | Specifies the user password                                                                                 | Not defined   |

## Security and encryption

| Parameter         | Description                                                                                                 | Default value |
|-------------------|-------------------------------------------------------------------------------------------------------------|---------------|
| `UseTls`          | Indicates whether to use the TLS protocol (`grpcs` or `grpc`)                                               | `false`       |

> | `RootCertificate` | Specifies the path to the trusted server TLS certificate. If this parameter is set, `UseTls` will be `true` | Not defined   |

Reviewer: Unclear: will it be true by default, or will it always be true and can't be overridden?

|-------------------|-------------------------------------------------------------------------------------------------------------|---------------|
| `MaxSessionPool`  | Specifies the maximum session pool size                                                                     | `100`         |

## Keepalive

| Parameter              | Description                                                                                                                                                                                                                                                                                                                             | Default value |
|------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------|

> | `KeepAlivePingDelay`   | The client will send a keep alive ping to the server if it doesn't receive any frames on a connection for this period of time. This property is used together with `KeepAlivePingTimeout` to check whether the connection is broken. Delay value must be greater than or equal to `1` second. Set to `0` to disable the keep alive ping | `10` seconds  |
> | `KeepAlivePingTimeout` | Keep alive pings are sent when a period of inactivity exceeds the configured `KeepAlivePingDelay` value. The client will close the connection if it doesn't receive any frames within the timeout. Timeout must be greater than or equal to `1` second. Set to `0` to disable the keep alive ping timeout                               | `10` seconds  |

Reviewer: ```suggestion
| `KeepAlivePingDelay`   | The client sends a keep-alive ping to the server if it doesn't receive any frames on a connection for this period of time. This property is used together with `KeepAlivePingTimeout` to check whether the connection is broken. The delay value must be greater than or equal to `1` second. Set to `0` to disable the keep-alive ping. | `10` seconds  |
| `KeepAlivePingTimeout` | Keep-alive pings are sent when a period of inactivity exceeds the configured `KeepAlivePingDelay` value. The client closes the connection if it doesn't receive any frames within the timeout. The timeout must be greater than or equal to `1` second. Set to `0` to disable the keep-alive ping timeout.                      | `10` seconds  |
```
